pyrometheus/__init__.py

```diff
@@ -95,7 +95,7 @@ def str_np_inner(ary):
 
 
 def str_np(ary):
-    return "np.array(%s)" % str_np_inner(ary)
```

Comment:
Agreed that in the non-MIRGE-Com use-case, this will result in a performance hit.

> Obviously, this can be fixed easily with an if statement, but the big question is what should go in the if statement, i.e. how do we recognize what use case we're in?

The codegen routine's caller could tell us the default `dtype` for all numpy array literals? Defaulted to whatever "auto" (i.e. let numpy infer). MIRGE-Com could just pass in `object` for that?

Reply:
Sure, that'd work. But is there a way for us to determine the use case without soliciting extra input from the user? (Such as by setting a default via `usr_np is np`?)

Reply:
That will be slightly ugly as `usr_np` is runtime dependent but the arrays literals are emitted while the `mako` template is being rendered. Allowed it to be set by the caller here: 710ea03, how does that look?